lib/ansible/modules/cloud/scaleway/scaleway_security_group.py

```diff
@@ -0,0 +1,240 @@
+  - name: Create a Security Group
+    scaleway_security_group:
+      state: present
+      region: par1
+      name: security_group
+      description: "my security group description"
+      organization: "43a3b6c8-916f-477b-b7ec-ff1898f5fdd9"
+      stateful: false
+      inbound_default_policy: accept
+      outbound_default_policy: accept
+      organization_default: false
+    register: security_group_creation_task
+'''
+
+RETURN = '''
+data:
+    description: This is only present when C(state=present)
+    returned: when C(state=present)
+    type: dict
+    sample: {
+        "scaleway_security_group": {
+            "description": "my security group description",
+            "enable_default_security": true,
+            "id": "0168fb1f-cc46-4f69-b4be-c95d2a19bcae",
+            "inbound_default_policy": "accept",
+            "name": "security_group",
+            "organization": "43a3b6c8-916f-477b-b7ec-ff1898f5fdd9",
+            "organization_default": false,
+            "outbound_default_policy": "accept",
+            "servers": [],
+            "stateful": false
+        }
+    }
+'''
+
+from ansible.module_utils.scaleway import SCALEWAY_LOCATION, scaleway_argument_spec, Scaleway
+from ansible.module_utils.basic import AnsibleModule
+from uuid import uuid4
+
+
+def payload_from_security_group(security_group):
+    return dict(
+        (k, v)
+        for k, v in security_group.items()
+        if k != 'id' and v is not None
+    )
+
+
+def present_strategy(api, security_group):
+    ret = {'changed': False}
+
+    response = api.get('security_groups')
+    if not response.ok:
+        api.module.fail_json(msg='Error getting security groups "%s": "%s" (%s)' % (response.info['msg'], response.json['message'], response.json))
+
+    security_groups = response.json['security_groups']
+    security_group_lookup = dict((sg['name'], sg)
+                                 for sg in security_groups)
+
+    if security_group['name'] not in security_group_lookup.keys():
+        ret['changed'] = True
+        if api.module.check_mode:
+            # Help user when check mode is enabled by defining id key
+            ret['scaleway_security_group'] = {'id': str(uuid4())}
+            return ret
+
+        # Create Security Group
+        response = api.post('/security_groups',
+                            data=payload_from_security_group(security_group))
+
+        if not response.ok:
+            msg = 'Error during security group creation: "%s": "%s" (%s)' % (response.info['msg'], response.json['message'], response.json)
+            api.module.fail_json(msg=msg)
+        ret['scaleway_security_group'] = response.json['security_group']
+
+    else:
+        ret['scaleway_security_group'] = security_group_lookup[security_group['name']]
+
+    return ret
+
+
+def absent_strategy(api, security_group):
+    response = api.get('security_groups')
+    ret = {'changed': False}
+
+    security_groups = response.json['security_groups']
```

Comment:
So, if the answer is not ok, how can we be sure that response.json will return something ?